### PR TITLE
changes for issue #77

### LINKED ISSUE #77: Change toast to snackbar messages
change all the toast messages in the app to snackbar messages.

### PR DESCRIPTION
**Pull request by git-typecode**
# Description
removed toast usage in the whole project.
added Snackbar where toasts were previously used
added a interface to implement in future classes that needs Snackbar.

Fixes #77

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

### CODE REVIEW

File: Android/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#Thu May 24 11:12:39 IST 2018
+#Tue Jul 03 11:13:27 CEST 2018
```
Comment (Swati4star) [inline Android/gradle/wrapper/gradle-wrapper.properties]: No need to make change in this file

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 2]
```diff
@@ -262,10 +257,12 @@ public void onResponse(Call call, Response response) throws IOException {
...
                 mHandler.post(() -> {
                     if (responseCode == STATUS_CODE_OK) {
-                        Toast.makeText(ProfileActivity.this, R.string.name_updated, Toast.LENGTH_SHORT).show();
+                        TravelmateSnackbars.createSnackBar(findViewById(R.id.activity_profile_id),
+                                R.string.name_updated, Snackbar.LENGTH_SHORT).show();
                         mSharedPreferences.edit().putString(USER_NAME, fullName).apply();
+
                     } else {
-                        Toast.makeText(ProfileActivity.this, res, Toast.LENGTH_LONG).show();
+                        displaySnackbar(res);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: Use travelmatesnackbar here also then.

File: Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java [hunk 3]
```diff
@@ -327,4 +324,10 @@ private void shareProfile() {
         }
     }
 
+    private void displaySnackbar(final String message) {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java]: No need of this function

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java [hunk 4]
```diff
@@ -173,9 +172,19 @@ private void hideProgressDialog() {
...
     }
 
+    /*
+    Old method for Toast messages. All should be replaced with Snackbar.
     private void displayToast(final String message) {
         mHandler.post(() -> Toast.makeText(getActivity(), message, Toast.LENGTH_SHORT).show());
     }
+    */
+
+    private void displaySnackbar(final String message) {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: Why is there need of this function? We have travelmatesnackbar class

Comment (git-typecode) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: Do you mean "interface TravelmateSnackbars"?

Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: I'll check the changes once locally also.

**Comment by Swati4star** [comment]
@git-typecode There are merge conflicts also. Resolve them.

**Comment by Swati4star** [comment]
By consistency @git-typecode , I mean  Android/app/src/main/java/io/github/project_travel_mate/ProfileActivity.java does not implements TravelmateSnackbars interface. It has standalone function for snackbars.  and many other files also. This is not consistent.

**Comment by git-typecode** [comment]
@Swati4star Could you help me to fix the issues from the Travis CI build? I dont get the issues if i do ./gradlew checkstyle. So i dont know how to fix it because it looks like my local repo dosent have the issues.

**Comment by Swati4star** [comment]
@git-typecode Are you sure, you merged the code properly?

**Comment by git-typecode** [comment]
I will revert to an old commit and sync the project from there and do it again.

File: .idea/codeStyles/Project.xml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (Swati4star) [inline .idea/codeStyles/Project.xml]: Remove the .idea folder from commit

**Comment by git-typecode** [comment]
@Swati4star will you merge it?

**Comment by Swati4star** [comment]
@git-typecode Really sorry! This was closed by mistake.

**Comment by Swati4star** [comment]
@git-typecode 
Please address the following comments & it will be ready for merge then
- Remove .idea folder from commit
- The Travis checks are failing
- This PR seems to have some other changes also (changes of other people or something that got messed up while merging the code)

File: Android/app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -76,6 +76,7 @@
     <string name="trip_added">Trip added</string>
     <string name="friend_added">Friend added to trip</string>
     <string name="no_friend_selected">Please select a valid friend to add</string>
+    <string name="images_added">Images added</string>
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: these shouldn't be a part of your change

File: Android/app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -104,8 +105,12 @@
     <string name="share_intent_text">Share Contact Details</string>
     <string name="share_menu">Share</string>
     <string name="share_contact_text">Share Contact</string>
+    <string name="share_contact">Share My Contact</string>
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: these strings also.

File: Android/app/src/main/res/layout/activity_my_trip_info.xml [hunk 8]
```diff
@@ -1,147 +1,159 @@
 <?xml version="1.0" encoding="utf-8"?>
-<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+<android.support.design.widget.CoordinatorLayout
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/activity_my_trip_info.xml]: The changes in this file should also not be a part of this PR

Comment (git-typecode) [reply Android/app/src/main/res/layout/activity_my_trip_info.xml]: Snackbars work best if they are displayed inside of a CoordinatorLayout so I added it and the ID to get the view.
TravelmateSnackbars.createSnackBar(findViewById(R.id.activityAddNewTrip),
R.string.trip_added, Snackbar.LENGTH_LONG).show();

File: Android/app/src/main/res/layout/activity_my_trip_info.xml [hunk 9]
```diff
@@ -1,147 +1,159 @@
...
+            <TextView
+                android:id="@+id/city_name"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
                 android:fontFamily="sans-serif-bold"
                 android:padding="10dp"
                 android:textColor="@color/black"
-                android:textSize="15sp"
-                android:visibility="invisible" />
+                android:textSize="25sp"
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/activity_my_trip_info.xml]: I meant these other changes in the file

File: Android/app/src/main/res/layout/activity_my_trip_info.xml [hunk 10]
```diff
@@ -1,147 +1,159 @@
...
-        <com.dd.processbutton.FlatButton
-            android:id="@+id/add_new_friend"
-            android:visibility="gone"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_margin="10dp"
-            android:background="@color/colorPrimary"
-            android:text="@string/prompt_add_friend"
-            android:textColor="@color/white" />
+                <TextView
```
Comment (Swati4star) [inline Android/app/src/main/res/layout/activity_my_trip_info.xml]: You have made a lot of changes in this file. Kind of removed a few views also

File: Android/app/src/main/res/values/strings.xml (hunk 7 above)
Comment (git-typecode) [reply Android/app/src/main/res/values/strings.xml]: In some old Toast messages the message String was typed directly in the makeText() so I replaced those with String rescources instead. You liked consistency right? 😃

Comment (Swati4star) [reply Android/app/src/main/res/values/strings.xml]: But you haven't used this string anywhere

File: Android/app/src/main/res/layout/activity_my_trip_info.xml (hunk 9 above)
Comment (git-typecode) [reply Android/app/src/main/res/layout/activity_my_trip_info.xml]: very strange, I havent edited that. I fix it.

**Comment by git-typecode** [comment]
🆗  now?

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/AddNewTripActivity.java [hunk 11]
```diff
@@ -60,13 +55,13 @@
...
         TimePickerDialog.OnTimeSetListener,
-        View.OnClickListener {
+        View.OnClickListener, TravelmateSnackbars {
 
     private static final String DATEPICKER_TAG1 = "datepicker1";
     @BindView(R.id.select_city_name)
     FlatButton cityName;
     @BindView(R.id.sdate)
-    FlatButton tripStartDate;
+    EditText tripStartDate;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/AddNewTripActivity.java]: why this change? It should be faltbutton only.

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/AddNewTripActivity.java [hunk 12]
```diff
@@ -129,9 +125,7 @@ public void onDateSet(DatePickerDialog datePickerDialog, int year, int month, in
         if (Objects.equals(datePickerDialog.getTag(), DATEPICKER_TAG1)) {
             Calendar calendar = new GregorianCalendar(year, month, day);
             mStartdate = Long.toString(calendar.getTimeInMillis() / 1000);
-            tripdate.setText(day + "/" + month + "/" + year);
-            tripdateText.setVisibility(View.VISIBLE);
-            tripdate.setVisibility(View.VISIBLE);
+            tripStartDate.setText(day + "/" + month + "/" + year);
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/AddNewTripActivity.java]: you have removed the two lines below this

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/MyTripInfoActivity.java [hunk 13]
```diff
@@ -10,6 +10,7 @@
 import android.os.Handler;
 import android.os.Looper;
 import android.preference.PreferenceManager;
+import android.support.design.widget.Snackbar;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/MyTripInfoActivity.java]: I don't think this import is required

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/MyTripsFragment.java [hunk 14]
```diff
@@ -9,24 +9,20 @@
 import android.os.Looper;
 import android.preference.PreferenceManager;
 import android.support.annotation.NonNull;
+import android.support.design.widget.Snackbar;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/mytrips/MyTripsFragment.java]: not needed

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java [hunk 15]
```diff
@@ -1,11 +1,11 @@
 package io.github.project_travel_mate.utilities;
 
-
 import android.content.SharedPreferences;
 import android.os.Bundle;
 import android.os.Handler;
 import android.os.Looper;
 import android.preference.PreferenceManager;
+import android.support.design.widget.Snackbar;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/BugReportFragment.java]: not needed

File: Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java [hunk 16]
```diff
@@ -9,30 +9,29 @@
...
-
 import java.io.File;
 import java.io.FileOutputStream;
 import java.util.Objects;
 import java.util.Random;
-
 import butterknife.BindView;
 import butterknife.ButterKnife;
 import io.github.project_travel_mate.R;
+import utils.TravelmateSnackbars;
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/utilities/ShareContactActivity.java]: not being used. not needed

File: Android/app/src/main/res/values/strings.xml (hunk 6 above)
Comment (Swati4star) [reply Android/app/src/main/res/values/strings.xml]: not yet addressed

File: Android/app/src/main/res/values/strings.xml [hunk 17]
```diff
@@ -105,8 +106,11 @@
     <string name="share_intent_text">Share Contact Details</string>
     <string name="share_menu">Share</string>
     <string name="share_contact_text">Share Contact</string>
+    <string name="cant_generate">Can\'t be generated. Need to download QR services</string>
+    <string name="add_contact">Added Contact</string>
+    <string name="cancel_contact">Cancelled Added Contact</string>
```
Comment (Swati4star) [inline Android/app/src/main/res/values/strings.xml]: the above three strings are not used anywhere.

Comment (git-typecode) [reply Android/app/src/main/res/values/strings.xml]: @Swati4star you need to be more specific. I check for unused strings in android studio and get no results. How do you know thay are unsused? Other people add code and i get conflicts that i need to resolve and merge my code. Then people add new toasts which i again resolve and merge.

File: Android/app/src/main/java/io/github/project_travel_mate/mytrips/MyTripsFragment.java (hunk 14 above)
Comment (git-typecode) [reply Android/app/src/main/java/io/github/project_travel_mate/mytrips/MyTripsFragment.java]: I guess it is used becouse of Snackbar.LENGTH_LONG
